integration_test/cloudbuild-integration-test.yaml

```diff
@@ -0,0 +1,137 @@
+﻿steps:
+- name: 'gcr.io/cloud-builders/git'
+  id: clone-dcm4che
+  args:
+  - clone
+  - https://github.com/dcm4che/dcm4che.git
+  - dcm4che
+
+- name: 'gcr.io/cloud-builders/git'
+  id: checkout-dcm4che-tag
+  args:
+  - checkout
+  - "tags/5.15.1"
+  dir: dcm4che
+
+- name: "maven:3.6.0-jdk-8"
+  id: build-store-scu
+  volumes:
+  - name: 'vol1'
+    path: '/root/.m2'
+  args:
+  - mvn
+  - install
+  dir: "dcm4che/dcm4che-tool/dcm4che-tool-storescu"
+
+# run adapter, close when connection established on _CLOSE_ADAPTER_PORT
+- name: 'gradle:5.4.1-jdk8'
+  id: gradle-run-adapter
+  args: [ 'bash', '../integration_test/run_adapter_and_wait.sh', '${_STAGE}', '${_PROJECT}', '${_LOCATION}', '${_DATASET}', '${_STORE_NAME}', '${_ADAPTER_PORT}', '${_CLOSE_ADAPTER_PORT}']
+  dir: import
+  waitFor: ['-']
+
+# delete dicom store if it already exists, ignore exit code
+- name: 'google/cloud-sdk:250.0.0'
+  id: cleanup-dicom-store
+  args:
+  - bash
+  - integration_test/store_delete_ignore_result.sh
+  - ${_LOCATION}
+  - ${_DATASET}
+  - ${_STORE_NAME}
+  waitFor: ['-']
+
+- name: 'google/cloud-sdk:250.0.0'
+  id: create-dicom-store
+  args:
+  - gcloud
+  - alpha
+  - healthcare
+  - dicom-stores
+  - create
+  - ${_STORE_NAME}
+  - "--dataset=${_DATASET}"
+  - "--location=${_LOCATION}"  
+  - "--quiet"
+  waitFor: 
+  - cleanup-dicom-store
+
+
+- name: 'busybox:latest'
+  id: wait-for-adapter
+  args:
+  - ash
+  - integration_test/wait_for_port.sh
```

Comment:
Instead of having the wait-for-adapter step, since run_adapter_and_wait.sh already waits for the adapter to start, do we need wait_for_port.sh or could we delete this step and add a waitFor in run-store-scu for build-store-scu, create-dicom-store, and gradle-run-adapter? You might have to modify run_adapter_and_wait.sh to wait for both the adapter port and the close port

Reply:
We can't use cloud build wait feature for gradle-run-adapter because I'm waiting for adapter to be up and running, not for it's step is finish.
So the solution I came up with is to run adapter, wait for it's dimse port to be up, run c-store scu, then use connection on close_port as signal to finish adapter cloud build step.